docs/02-guides/customize-wysiwyg-platform.mdx

```diff
@@ -0,0 +1,200 @@
+
+- `<a>` tags are transformed into `theme/components/atoms/Typography/Link`
+  components when the `href` attribute does not contain a domain.
+- `<widget>` tags are transformed into
+  `theme/modules/Wysiwyg/MagentoWysiwyg/Widget/Widget.js` components. However,
+  you shouldn't write a `<widget>` tag manually. It comes from the
+  `{% raw %}{{widget}}{% endraw %}` shortcode.
+- `<style>` tags are transformed to support
+  [Magento's Page Builder](/docs/3.x/extensions/magento2/how-to/use-and-extend-the-page-builder)
+  format. The `#html-body` selector is replaced with the `<Wysiwyg>` root
+  selector. See
+  [WYSIWYG dynamic styles](/docs/3.x/guides/display-wysiwyg-content#dynamic-styles)
+  for details.
+
+### Add a custom Magento Widget
+
+Custom widgets will be automatically parsed. However, you will still need to map
+the widget's `type` to custom React Components. If you don't, they will be
+ignored and nothing will be rendered in the final page.
+
+To do so, please override
+`theme/modules/Wysiwyg/MagentoWysiwyg/Widget/getWidgetComponent.js` in your own
+theme.
+
+Now, within your newly created `getWidgetComponent.js` file, you will be able to
+add your own behavior. For instance, if you've created a widget in Magento that
+uses the type `acme/product-preview`, you will need to update the `Widget.js`
+file likewise:
+
+```js title="theme/modules/Wysiwyg/MagentoWysiwyg/Widget/getWidgetComponent.js"
+import React from "react";
+import Spinner from "theme/components/atoms/Spinner";
+import loadable from "@loadable/component";
+
+const defaultWidgetsMap = {
+  "acme/product-preview": loadable(
+    () => import("./path/to/ProductPreview.js"),
+    {
+      fallback: <Spinner />,
+    }
+  ),
+};
+
+// ... rest of the file is intact
+```
+
+If you restart your application, you will notice that the new widget component
+is displayed. Hurray!
+
+However, in most cases, you will need to fetch data to display all the needed
+information in your widget. For instance, if the widget is
+`{% raw %}{{ widget type="acme/product-preview" sku="VSK12" }}{% endraw %}` you
+will want to fetch the product associated with the given SKU.
+
+#### Register your widget type at the GraphQL level
+
+1.  First make sure that your dependencies are up to date in your GraphQL module
+    ```ts title="extensions/acme-extension/modules/wysiwyg/index.ts"
+    export default createGraphQLModule({
+    namespace: "Magento2/Cms",
+    // highlight-start
+    dependencies: [
+      "Magento2/Wysiwyg", // ensure that Widget related features are available
+      "Magento2/Catalog/Product" // ensure that you can fetch a product in your Wysiwyg data
+    ],
+    // highlight-end
+    ```
+1.  Then you need to register a new WidgetData type in your schema
+    ```ts title="extensions/acme-extension/modules/wysiwyg/index.ts"
+    typeDefs: /* GraphQL */ `
+      type WidgetProductPreviewData implements WidgetData {
+        dataId: ID
+        product: Product
+      }
+    `,
```

Comment:
we now inline the `typeDefs` it should be in the future ADR for https://linear.app/front-commerce/issue/FC-2044/[vite]-refactor-graphql-modules